src/DataReader/parse_queries.rs

```diff
@@ -172,6 +175,52 @@ fn build_expression_from_pair(pair: pest::iterators::Pair<Rule>) -> QueryExpress
     }
 }
 
+fn build_state_from_pair(pair: pest::iterators::Pair<Rule>) -> QueryExpression {
+    let mut inner_pair = pair.clone().into_inner();
+    let locPair = inner_pair.next().unwrap();
+
+    let mut loc_names: Vec<Box<QueryExpression>> = Vec::new();
+    for loc_name in locPair.as_str().split(',') {
+        loc_names.push(Box::new(QueryExpression::LocName(
+            loc_name.trim().to_string(),
+        )));
+    }
+
+    let clock_pair = inner_pair.next().unwrap();
+
+    // In the following line of code, we build a BoolExprssion based on the clock constraints defined for the given location.
+    // To make BoolExprssion we use the InvariantParser parser instead.
+    // Becuase clocks is defined as c1&&c2... in the InvariantParser we replace ',' to match the format e.g., e.g., "x>0,y<5" => "x>0&&y<5"
+    let invariant_version: Option<Box<BoolExpression>> = if clock_pair.as_str().trim() != "" {
+        let clock_string = clock_pair.as_str().trim().to_string().replace(',', "&&");
+        let invariant_version = parse(&clock_string).expect("");
+        Some(Box::new(invariant_version))
+    } else {
+        None
+    };
```

Comment:
We should either update the invariant grammar to include diagonal constraints or use the edge grammar guard instead here. We **need** diagonal constraints here for the feature to be useful during simulation. 

We should eventually combine the invariant grammar and edge grammar: the edge guard should be the equivalent to the invariant and it would simplify our parser code